### DISCUSSION TITLE
Simulation freezes in the first seconds

### DISCUSSION THREAD

**Discussion by MJavadZallaghi** [cyberbotics/wrestling#77] [Q&A] [answered]
Dear @omichel, Hi; Hope you are doing well.

Sometimes in the first seconds of a competition, while the global timer is started, the robots are frozen (one or both). As in some control and motion strategies the access to the real timer is important, and sometimes the latency of the control loop causes inaccurate manual time counting, it is very important to have access to the real timer from the start of the motion not from the overall starts, which includes the mentioned motion freezing phase.

Do you have any plan to prevent the explained freezing phase? How can we as participants prevent the mentioned situation?

Sometimes in the first run, I have the same situation on my own computer, but after the resetting, I don't have any freeze-mode in the next loops of simulation.

Thanks in advance for your helps.

Best regards,
Mohammad Javad Zallaghi

**Comment by christopherpeers** [comment]
When your code begins, take the self.time once and then offset that value from self.time whenever you want to use the actual time.

**Comment by omichel** [comment, marked as answer]
I don't know exact where does this freezing time come from. Probably it is a problem with the initialization of the docker container which takes some time or the Linux scheduler which randomly favors the Webots docker instead of the robot controller dockers. I am not sure how to fix this... But @christopherpeers is right: using `self.time` will give you the actual time of the simulation.
